tokio/src/io/async_fd.rs

```diff
@@ -85,14 +85,27 @@ impl<T: AsRawFd> AsyncFd<T> {
     where
         T: AsRawFd,
     {
-        Self::new_with_handle(inner, Handle::current())
+        Self::new_with_handle_and_interest(inner, Handle::current(), ALL_INTEREST)
     }
 
-    pub(crate) fn new_with_handle(inner: T, handle: Handle) -> io::Result<Self> {
+    /// Creates new instance as `new` with additional ability to customize interest,
+    /// allowing to specify whether file descriptor will be polled for read, write or both.
+    pub fn new_with_interest(inner: T, interest: Interest) -> io::Result<Self>
```

Comment:
Thanks, I'm wondering if this should just be `with_interest` (modeling after `Vec::with_capacity`).

The other `new_with` fns are internal and haven't received as much attention naming wise.

Thoughts @tokio-rs/maintainers ?

Reply:
i have a slight preference for `with_interest`, but i could be convinced otherwise.

Reply:
Changed it to `with_interest`, it is shorter too so I would prefer it